File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/ConnectedEvent.java
package dev.robocode.tankroyale.botapi.events;

import java.net.URI;

/**
 * Event occurring when bot gets connected to server
 */
public final class ConnectedEvent extends ConnectionEvent {

    /**
     * Initializes a new instance of the ConnectedEvent class.
     *
     * @param serverUri is the URI of the server.
     */
    public ConnectedEvent(URI serverUri) {
        super(serverUri);
    }
}

File: bot-api/dotnet/Robocode.TankRoyale.BotApi/src/events/ConnectedEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events;

/// <summary>
/// Event occurring when bot gets connected to server.
/// </summary>
public sealed class ConnectedEvent : ConnectionEvent
{
    /// <summary>
    /// Initializes a new instance of the ConnectedEvent class.
    /// </summary>
    /// <param name="serverUri">URI of the server.</param>
    [JsonConstructor]
    public ConnectedEvent(System.Uri serverUri) : base(serverUri)
    {
    }
}